### PR TITLE
Return empty array if no redirects are find for a given pageType

### PR DESCRIPTION
**Pull request by hakubo**
Previously if you wanted redirects for 'hubsv3' and it was not in the RedirectPages,
it would return the whole redirects list.

### CODE REVIEW

**Comment by hakubo** [comment]
This should actually brake the tests, 
who knows how this should behave?

**Comment by hakubo** [comment]
@kvas-damian @SebastianMarzjan do you know how this should behave?
This fixes problem with Title.php, but will it brake anything else?

File: includes/wikia/services/RedirectService.class.php [hunk 1]
```diff
@@ -82,11 +83,17 @@ public function getRedirects() {
...
-		if ( isset( $this->pageType )
-			&& isset( $this->redirects[ $this->pageType ] )
-			&& is_array( $this->redirects[ $this->pageType ]) )
-		{
-			return $this->redirects[ $this->pageType ];
+
+		if ( isset( $this->pageType ) ) {
+			if (
+				isset( $this->redirects[ $this->pageType ] )
+				&& is_array( $this->redirects[ $this->pageType ])
```
Comment (bognix) [inline includes/wikia/services/RedirectService.class.php]: you first check if key is set and then you check if this is an array, why not other way around? If an array, then check the key.

**Comment by kvas-damian** [comment]
@hakubo redirects format is decribed here: https://github.com/Wikia/app/blob/dev/includes/wikia/services/RedirectService.class.php#L3

If there is no ```pageType``` key then we use entire array as ```old_url => new_url``` mapping

File: includes/wikia/services/RedirectService.class.php (hunk 1 above)
Comment (kvas-damian) [inline includes/wikia/services/RedirectService.class.php]: @bognix you need to make sure that this key is present to check if it's value is an array.

Comment (bognix) [inline includes/wikia/services/RedirectService.class.php]: aaa sorry! I misread...

Comment (hakubo) [inline includes/wikia/services/RedirectService.class.php]: if I check ['a'][1] this will give me a notice: Notice: Undefined offset

**Comment by hakubo** [comment]
@kvas-damian does that sound good now?

File: includes/wikia/services/RedirectService.class.php [hunk 2]
```diff
@@ -82,11 +78,17 @@ public function getRedirects() {
 			$this->redirects = $this->prepareRedirects();
 		}
 
-		if ( isset( $this->pageType )
-			&& isset( $this->redirects[ $this->pageType ] )
-			&& is_array( $this->redirects[ $this->pageType ]) )
-		{
-			return $this->redirects[ $this->pageType ];
+
+		if ( !empty( $this->pageType ) ) {
```
Comment (kvas-damian) [inline includes/wikia/services/RedirectService.class.php]: Does it still make sense to keep this condition separately?

Comment (hakubo) [inline includes/wikia/services/RedirectService.class.php]: I think it improves readability

Comment (kvas-damian) [inline includes/wikia/services/RedirectService.class.php]: ok, lets keep it then.

**Comment by kvas-damian** [comment]
LGTM

**Comment by tomasznapieralski** [comment]
All automated WamPageTest passed

**Pull request merged**